src/main/java/core/basesyntax/converter/StringToFruitDaoConverter.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.dao.FruitDao;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class StringToFruitDaoConverter implements Function<String, FruitDao> {
+
+    @Override
+    public FruitDao apply(String s) {
+        String[] arr = s.split(",");
+        if (arr.length != 3) {
+            throw new RuntimeException("Cannot convert string [" + s + "] to fruitDao");
+        }
+        return new FruitDao(arr[1], Integer.parseInt(arr[2]));
```

Comment:
Ensure that the `FruitDao` constructor is correctly handling the input parameters. Since `FruitDao` is expected to represent a fruit with a name and quantity, verify that the input data is valid and meaningful.